### PR TITLE
Fix numpy bug

### LINKED ISSUE #123: NumPy bug with newest version
The newer versions of NumPy have deprecated the ability to use a `slice` object as an index, which breaks the code at these lines: https://github.com/biosimulators/Biosimulators_utils/blob/9355badea9324b57a23330b57becf85fa2a37ace/biosimulators_utils/report/io.py#L270-L271

This is the resulting error message:
```
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

This code needs to be reworked to index into this `results` NumPy array with variable number of dimensions and index ranges.

### PR DESCRIPTION
**Pull request by eagmon**
This PR was previously opened with a different branch in PR #121. This PR fixes a long-standing NumPy error in which a slice object can no longer be used as an index, described in issue #123. Instead, I am pulling the ranges out of the slice object and using them directly.

I also fixed some listing errors and matplotlib deprecation errors, but left 4 tests still failing, down from 13 failing before this PR:

FAILED tests/model_lang/bngl/test_bngl_utils.py::BgnlUtilsTestCase::test_get_parameters_variables_for_simulation_with_empty_sample_times - AssertionError: “failed to parse action” does not match “Model file /home/runner/work/Biosimulators_utils/Biosimulators_utils/tests/model_lang/bngl/../../fixtures/bngl/empty-sample-times.bngl is not a valid BNGL or BNGL XML file.
FAILED tests/sedml/test_sedml_validation.py::ValidationTestCase::test_validate_calculation - AssertionError: ‘The syntax’ not found in ‘- The mathematical expression a * is invalid.\n - unexpected EOF while parsing (, line 1)’
FAILED tests/combine/test_combine_exec.py::ExecCombineTestCase::test_1 - AssertionError: None != {‘sim.sedml’: {‘report1’: ‘ABC’, ‘report2’: ‘DEF’}}
FAILED tests/combine/test_combine_exec.py::ExecCombineTestCase::test_2 - AssertionError: Lists differ: [‘log.yml’] != [‘dir1’, ‘log.yml’, ‘plots.zip’, ‘reports.zip’]

### CODE REVIEW

**Comment by CodeByDrescher** [comment]
I found an alternate solution to fix the numpy bug; turns out multi-dimensional indexing still works if its done using a tuple. We don't need the slices to be mutable, so a cast to tuple worked just fine for our purposes. Closes #123

**Comment by eagmon** [comment]
nice find @CodeByDrescher! Looks like the NumPy bug is fixed. There are still 4 tests failing, but maybe those should be addressed in a separate PR?

**Review by jcschaff — APPROVED** [review]
approved, I like the tuple solution for numpy.

**Review by jcschaff — CHANGES_REQUESTED** [review]
can we fix the CI actions?

**Comment by eagmon** [comment]
@jcschaff -- there are 4 failing tests left. Maybe we divide and conquer? We could also do that in a separate PR if we just want to get the NumPy bug fixed now to unblock progress in the test suite.

File: biosimulators_utils/combine/validation.py [hunk 1]
```diff
@@ -22,14 +23,14 @@
...
 
 
-def validate(archive, archive_dirname,
-             include_all_sed_docs_when_no_sed_doc_is_master=True,
-             always_include_all_sed_docs=False,
-             formats_to_validate=[
+def validate(archive:CombineArchive, archive_dirname:str,
+             include_all_sed_docs_when_no_sed_doc_is_master:bool=True,
+             always_include_all_sed_docs:bool=False,
+             formats_to_validate:"list[CombineArchiveContentFormat]"=[
```
Comment (eagmon) [inline biosimulators_utils/combine/validation.py]: what kind of type is this? a string?

File: biosimulators_utils/combine/io.py [hunk 2]
```diff
@@ -201,7 +202,11 @@ def run(self, in_file, out_dir, include_omex_metadata_files=True, config=None):
             archive.contents.append(content)
 
         # extract files
-        archive_comb.extractTo(out_dir)
+
+        with Zip(in_file, 'r') as zip_archive:
+            zip_archive.extractall(path=out_dir)
```
Comment (jcschaff) [inline biosimulators_utils/combine/io.py:207]: reminder to send bug report to  libcombine

File: docs-src/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 pydata-sphinx-theme
 sphinx >= 1.8
-sphinxprettysearchresults
+sphinxprettysearchresults @ git+https://github.com/biosimulators/sphinx-pretty-searchresults.git
```
Comment (jcschaff) [inline docs-src/requirements.txt:3]: add issue to sphinxprettysearchresults to add an import os (so we don't need the fork).

**Review by jcschaff — APPROVED** [review]
Great job, it works.  Logan discovered issues with two libraries and sent/will send notes to the authors.

**Pull request merged**